### ISSUE TITLE
Nested sorting with foreign key on :related not possible?

### ISSUE THREAD

**Issue by fsauter** [JSONAPI-Resources/jsonapi-resources#1086]
Hi,

I try to sort a nested attribute with a has one (where the foreign key can be found on the related resource):

```ruby
class UserResource < Api::V1::BaseResource
  has_one :profile, foreign_key_on: :related
  def self.sortable_fields(context)
    super + [:'profile.first_name']
  end
end

class ProfileResource < Api::V1::BaseResource
end
```
A call to `users?include=profile&sort=profile.first_name` results in this query:
`... LEFT JOIN profiles AS profile_sorting ON profile_sorting.id = users.user_id ORDER BY profile_sorting.first_name asc LIMIT 10 OFFSET 0`

My problem: The foreign key is on profile - so this would be correct:
`... LEFT JOIN profiles AS profile_sorting ON profile_sorting.user_id = users.id ORDER BY profile_sorting.first_name asc LIMIT 10 OFFSET 0`

I've already checked [_build_joins:resource.rb#L669](https://github.com/cerebris/jsonapi-resources/blob/release-0-8/lib/jsonapi/resource.rb#L669) where the join is generated and it seems to be hard coded. I would expect that :foreign_key_on on the relation set to :related would fix the problem. But its not considered while building the join. Is this by purpose? Is there another way to solve this problem or do I have to override the `_build_joins` method?

Greetings Florian

PS: Just as reference, this issue may relates to the functionality added here: #494

**Comment by fsauter** [comment]
I "quickfixedit" now for me like this (temporarily). I guess that's not the best solution.. Would still love to have the "best way" answer?

  ```ruby
class UserResource < Api::V1::BaseResource

  has_one :profile

  def self.sortable_fields(context)
    super + [:'profile.first_name']
  end

  def self._build_joins(associations)
    joins = []

    associations.inject do |prev, current|
      if current.class == ActiveRecord::Reflection::HasOneReflection
        joins << "LEFT JOIN #{current.table_name} AS #{current.name}_sorting ON #{current.name}_sorting.#{current.foreign_key} = #{prev.table_name}.id"
      else
        joins << "LEFT JOIN #{current.table_name} AS #{current.name}_sorting ON #{current.name}_sorting.id = #{prev.table_name}.#{current.foreign_key}"
      end
      current
    end
    joins.join("\n")
  end

end
```

**Comment by senid231** [comment]
@fsauter I think I have fixed this in #1112
